packages/eslint-plugin/src/utils/runtime-lint-utils.ts

```diff
@@ -103,6 +103,12 @@ function hasTag(proj: ProjectGraphProjectNode, tag: string): boolean {
     return (proj.data.tags || []).some((t) => regex.test(t));
   }
 
+  // if the tag is a glob, check if the project matches the glob prefix
+  if (tag.endsWith('*')) {
+    const prefix = tag.substring(0, tag.length - 1);
+    return (proj.data.tags || []).some((t) => t.startsWith(prefix));
+  }
```

Comment:
I know this is a pretty naïve implementation of glob support, but since this option already supports regex I figured it didn't really need to be ultra robust. Happy to revisit if desired